dluhc-component-library/src/components/timetablePage/TimetablePage.tsx

```diff
@@ -1,20 +1,37 @@
 import { useEffect, useState } from "preact/hooks";
 import Timetable from "./Timetable";
+import csvToJson from "csvtojson";
 
 interface TimetablePageProps {
   filepath: string;
 }
 
-const loadData = async (filepath: string) =>
+enum fileEndings { 
+  CSV= ".csv", 
+  JSON = ".json"
+}
+
+const loadJson = async (filepath: string) =>
   await fetch(filepath).then((res) => res.json());
 
-const TimetablePage = ({ filepath }: TimetablePageProps) => {
+const loadCSV = async (filepath: string) =>
+  await fetch(filepath).then((res) => res.text());
+
+const TimetablePage = ( { filepath }: TimetablePageProps) => {
   const [timetableData, setTimetableData] = useState();
 
   useEffect(() => {
-    loadData(filepath).then((data) => {
-      setTimetableData(data);
-    });
+    if (filepath.includes(fileEndings.CSV)) {
```

Comment:
Not sure if I should get rid of the enum and just have an `if/else` instead of `if/else if `